sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeViewModel.kt

```diff
@@ -38,15 +38,19 @@ class HomeViewModel @Inject constructor(
             .catch { e -> _error.emit(e) }
             .launchIn(viewModelScope)
 
-        launch(dispatchersProvider.io) {
-            val isFirstTimeLaunch = isFirstTimeLaunchPreferencesUseCase()
-                .catch { e -> _error.emit(e) }
-                .first()
-
-            _isFirstTimeLaunch.emit(isFirstTimeLaunch)
-            if (isFirstTimeLaunch) {
-                updateFirstTimeLaunchPreferencesUseCase(false)
+        isFirstTimeLaunchPreferencesUseCase()
+            .onEach { isFirstTimeLaunch ->
+                _isFirstTimeLaunch.emit(isFirstTimeLaunch)
             }
+            .flowOn(dispatchersProvider.io)
+            .catch { e -> _error.emit(e) }
+            .launchIn(viewModelScope)
```

Comment:
Need to `catch` before `onEach` or it will throw I think.
```suggestion
        isFirstTimeLaunchPreferencesUseCase()
            .catch { e -> _error.emit(e) }
            .onEach { isFirstTimeLaunch ->
                _isFirstTimeLaunch.emit(isFirstTimeLaunch)
            }
            .flowOn(dispatchersProvider.io)
            .launchIn(viewModelScope)
```

Reply:
The unit test for the error case seems to be working fine with the current code: https://github.com/nimblehq/android-templates/pull/478/commits/29ccca967a3952f212823388c49c57053600e98a